strongbox-user-management/src/main/java/org/carlspring/strongbox/users/security/SecurityTokenProvider.java

```diff
@@ -0,0 +1,128 @@
+package org.carlspring.strongbox.users.security;
+
+import java.security.Key;
+import java.util.Date;
+import java.util.Map;
+
+import org.carlspring.strongbox.security.exceptions.SecurityTokenException;
+import org.jose4j.jws.AlgorithmIdentifiers;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.jwt.JwtClaims;
+import org.jose4j.jwt.MalformedClaimException;
+import org.jose4j.jwt.consumer.InvalidJwtException;
+import org.jose4j.jwt.consumer.JwtConsumer;
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+import org.jose4j.keys.HmacKey;
+import org.jose4j.lang.JoseException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Used to get and verify security tokens. <br>
+ * This implementation based on JSON Web Token (JWT) which is RFC 7519 standard.
+ * <br>
+ * 
+ * 
+ * @author Sergey Bespalov
+ */
+@Component
+public class SecurityTokenProvider
+{
+    /**
+     * Secret key which is used to encode and verify tokens.<br>
+     * All previous tokens will be invalid, if it changed.
+     */
+    private Key key;
+
+    /**
+     * Creates {@link Key} instance using Secret string from application
+     * configuration.
+     * 
+     * @param secret
+     * @throws Exception
+     */
+    @Autowired
+    public void init(@Value("${strongbox.security.jwtSecret:secret}") String secret)
```

Comment:
I assume a good place to store this would be in the `Configuration` class...? (`strongbox.xml`)